### PR TITLE
e2e: TestLeaderElection: delete the lease and increase the retry period

### PR DESCRIPTION
**Pull request by ingvagabund**
In some random cases none of the deschedulers acquires a lease. Making the test fail due to already existing lease.

### CODE REVIEW

File: test/e2e/e2e_leaderelection_test.go [hunk 1]
```diff
@@ -141,7 +156,11 @@ func TestLeaderElection(t *testing.T) {
 			t.Logf("Only the pods in %s namespace are evicted. Pods before: %s, Pods after %s", ns2, podListBOrg, podListB)
 		}
 	} else {
-		t.Fatalf("Pods are evicted in both namespaces. For %s namespace Pods before: %s, Pods after %s. And, for %s namespace Pods before: %s, Pods after: %s", ns1, podListAOrg, podListA, ns2, podListBOrg, podListB)
+		if left && right {
+			t.Fatalf("No pods evicted. Probably none of the deschedulers were running.")
```
Comment (JaneLiuL) [inline test/e2e/e2e_leaderelection_test.go:160]: should it change int t.Lof?

Comment (ingvagabund) [reply test/e2e/e2e_leaderelection_test.go:160]: The test needs to fail here as well. The test expect at least one descheduler running.

Comment (JaneLiuL) [reply test/e2e/e2e_leaderelection_test.go:160]: Oh, ok, then lgtm now~~ thanks for your explain~

**Review by JaneLiuL — COMMENTED** [review]
/lgtm

File: test/e2e/e2e_leaderelection_test.go [hunk 2]
```diff
@@ -95,9 +99,20 @@ func TestLeaderElection(t *testing.T) {
...
+	}
 	s2.ClientConnection.Kubeconfig = os.Getenv("KUBECONFIG")
 	s2.PolicyConfigFile = "./policy_leaderelection_b.yaml"
 
+	// Delete the descheduler lease
+	err = clientSet.CoordinationV1().Leases("kube-system").Delete(ctx, "descheduler", metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		t.Fatalf("Unable to remove kube-system/descheduler lease: %v", err)
+	} else {
+		t.Logf("Removed kube-system/descheduler lease")
```
Comment (a7i) [inline test/e2e/e2e_leaderelection_test.go:113]: a bit misleading since not found doesn't actually remove

Comment (ingvagabund) [reply test/e2e/e2e_leaderelection_test.go]: You are right. Updated. Thx.

**Pull request merged**